src/Discovery/index.tsx

```diff
@@ -60,42 +60,49 @@ const DiscoveryWithMDSBackend: React.FC<{
-        if (!useArboristUI) {
-          throw new Error('Arborist UI must be enabled for the Discovery page to work if authorization is enabled in the Discovery page. Set `useArboristUI: true` in the portal config.');
-        }
-        const studiesWithAccessibleField = rawStudies.map((study) => {
-          let accessible: AccessLevel;
-          if (study[authzField] === undefined || study[authzField] === '') {
-            accessible = AccessLevel.NOTAVAILABLE;
-          } else {
-            accessible = userHasMethodForServiceOnResource('read', '*', study[authzField], props.userAuthMapping)
-              ? AccessLevel.ACCESSIBLE
-              : AccessLevel.UNACCESSIBLE;
-          }
-          return {
-            ...study,
-            __accessible: accessible,
-          };
-        });
-        setStudies(studiesWithAccessibleField);
+    async function loadStudiesWrapper() {
+      let loadStudiesFunction;
+      let getUserAuthMapping;
+      if (isEnabled('discoveryUseAggMDS')) {
+        loadStudiesFunction = loadStudiesFromAggMDS;
+        const userAuthMappings = await loadAuthMappingsFromWTS();
+        getUserAuthMapping = study => userAuthMappings[study.commons_url];
       } else {
-        setStudies(rawStudies);
+        loadStudiesFunction = loadStudiesFromMDS;
+        getUserAuthMapping = () => props.userAuthMapping;
       }
-    }).catch((err) => {
-      // eslint-disable-next-line no-console
-      console.error('Error encountered while loading studies: ', err);
-    });
+      loadStudiesFunction().then((rawStudies) => {
+        if (props.config.features.authorization.enabled) {
+          // mark studies as accessible or inaccessible to user
+          const { authzField } = props.config.minimalFieldMapping;
+          // useArboristUI=true is required for userHasMethodForServiceOnResource
+          if (!useArboristUI) {
+            throw new Error('Arborist UI must be enabled for the Discovery page to work if authorization is enabled in the Discovery page. Set `useArboristUI: true` in the portal config.');
+          }
+          const studiesWithAccessibleField = rawStudies.map((study) => {
+            let accessible: AccessLevel;
+            if (study[authzField] === undefined || study[authzField] === '') {
+              accessible = AccessLevel.NOTAVAILABLE;
+            } else {
+              accessible = userHasMethodForServiceOnResource('read', '*', study[authzField], getUserAuthMapping(study))
+                ? AccessLevel.ACCESSIBLE
+                : AccessLevel.UNACCESSIBLE;
+            }
+            return {
+              ...study,
+              __accessible: accessible,
+            };
+          });
+          setStudies(studiesWithAccessibleField);
+        } else {
+          setStudies(rawStudies);
+        }
+      }).catch((err) => {
+        // eslint-disable-next-line no-console
+        console.error('Error encountered while loading studies: ', err);
```

Comment:
Minor point, but JS automatically adds a space after `,` separated values.
```suggestion
        console.error('Error encountered while loading studies:', err);
```